test-env/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "test-env"
```

Comment:
this name is the crate name on crates.io, so it needs to be `restate-sdk-test-env` (unlike the directory name, which `test-env` is just fine)